### PR TITLE
refactor: remove unstable feature const_trait_impl & const_slice_index & slice_group_by

### PR DESCRIPTION
**Pull request by suica**
for #626

1. Removed `const_trait_impl` & `const_slice_index`.
2. Removed `slice_group_by`.

### CODE REVIEW

**Review by Boshen — APPROVED** [review]

**Review by Boshen — CHANGES_REQUESTED** [review]
Need a quick `cargo fmt`?

**Comment by Boshen** [comment]
Can I get separate PRs after this one? So we manage this more easily.

File: crates/oxc_parser/src/lexer/simd.rs [hunk 1]
```diff
@@ -39,23 +39,25 @@ impl SkipWhitespace {
...
 
     pub fn simd(&mut self, bytes: &[u8]) -> &Self {
-        let (chunks, remainder) = bytes.as_chunks::<ELEMENTS>();
+        let chunks = bytes.chunks(ELEMENTS);
 
         for chunk in chunks {
-            self.check_chunk(chunk);
-            if self.found {
-                return self;
+            if chunk.len() != ELEMENTS {
```
Comment (suica) [inline crates/oxc_parser/src/lexer/simd.rs:45]: @Boshen For the unstable `slice.as_chunks` api, `chunks` in `(chunks, remainder)` could also have a length less than `ELEMENTS`, if `bytes.len()<ELEMENTS`. 

So, there could be chunk with a length less than `ELEMENTS` go into this for loop. Is that as expected?

**Comment by suica** [comment]
@Boshen Ok, I'll leave the `slice_as_chunks`, and make another PR.

File: crates/oxc_parser/src/lexer/simd.rs (hunk 1 above)
Comment (suica) [reply crates/oxc_parser/src/lexer/simd.rs:45]: Move to another PR.

**Comment by suica** [comment]
> @Boshen Ok, I'll leave the `slice_as_chunks`, and make another PR.

`slice_as_chunks` is removed in https://github.com/web-infra-dev/oxc/pull/632.

**Review by Boshen — APPROVED** [review]

**Pull request merged**